sdk/common/src/main/scala/org/typelevel/otel4s/sdk/Attributes.scala

```diff
@@ -92,15 +108,112 @@ object Attributes {
   }
 
   implicit val hashAttributes: Hash[Attributes] =
-    Hash.by(_.m)
+    Hash.by(_.toMap)
 
   implicit val monoidAttributes: Monoid[Attributes] =
     new Monoid[Attributes] {
       def empty: Attributes = Attributes.Empty
-      def combine(x: Attributes, y: Attributes): Attributes = {
+      def combine(x: Attributes, y: Attributes): Attributes =
         if (y.isEmpty) x
         else if (x.isEmpty) y
-        else new Attributes(x.m ++ y.m)
+        else new MapAttributes(x.toMap ++ y.toMap)
+    }
+
+  /** A '''mutable''' builder of [[Attributes]].
+    */
+  final class Builder extends mutable.Builder[Attribute[_], Attributes] {
+    private val builder = Map.newBuilder[AttributeKey[_], Attribute[_]]
+
+    /** Adds the attribute with the given `key` and `value` to the builder.
+      *
+      * @note
+      *   if the given `key` is already present in the builder, the value will
+      *   be overwritten with the given `value`.
+      *
+      * @param key
+      *   the key of the attribute. Denotes the types of the `value`
+      *
+      * @param value
+      *   the value of the attribute
+      */
+    def addOne[A](key: AttributeKey[A], value: A): this.type = {
```

Comment:
should there also be on of these that takes a string name and an implicit `KeySelect`?

Reply:
I like the idea